src/engraving/playback/playbackcontext.cpp

```diff
@@ -139,9 +139,15 @@ void PlaybackContext::update(const ID partId, const Score* score)
                 int segmentStartTick = segment->tick().ticks() + tickPositionOffset;
 
                 for (const EngravingItem* item : segment->elist()) {
```

Comment:
For scores with many parts, unnecessarily looping over the whole `elist` seems inefficient to me. Instead, I'd do something like this:
```c++
    const Part* part = …; // get part belonging to partId
    const track_idx_t startTrack = part->startTrack();
    const track_idx_t endTrack = part->endTrack();

    …
            for (Segment* segment = measure->first(); segment; segment = segment->next()) {
                 int segmentStartTick = segment->tick().ticks() + tickPositionOffset;
                 
                 for (track_idx_t track = startTrack; track < endTrack; ++track) {
                     const EngravingItem* item = segment->elementAt(track);
                     if (!item || !item->isMeasureRepeat()) { continue; }
                     …
```

Reply:
Thanks, good idea! Fixed